### PR TITLE
Init xds server for APK Management server

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
## Purpose
Fixes: https://github.com/wso2/apk/issues/25

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: management-server/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"os"
+	"os/signal"
+
+	"github.com/wso2/apk/APKManagementServer/internal/logger"
+	"github.com/wso2/apk/APKManagementServer/internal/xds"
+)
+
+func main() {
+	logger.LoggerServer.Info("Hello, world.")
```
Comment (VirajSalaka) [inline management-server/cmd/main.go:29]: `Starting Management Server ...`

File: management-server/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org).
```
Comment (AmaliMatharaarachchi) [inline management-server/cmd/main.go:2]:
```suggestion
 *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org).
```

File: management-server/go.mod [hunk 3]
```diff
@@ -0,0 +1,27 @@
+module github.com/wso2/apk/APKManagementServer
+
+go 1.18
```
Comment (AmaliMatharaarachchi) [inline management-server/go.mod:3]:
```suggestion
go 1.19
```

**Review by VajiraPrabuddhaka — DISMISSED** [review]

File: management-server/internal/config/parser.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+var (
+	onceConfigRead         sync.Once
+	mgwHome                string
+	managementServerConfig *Config
+)
+
+//constants related to utility functions
+const (
+	// RelativeConfigPath is the relative file path where the configuration file is.
+	relativeConfigPath = "/conf/ga_config.toml"
```
Comment (VajiraPrabuddhaka) [inline management-server/internal/config/parser.go:40]: shall we rename as `mgt_config.toml` or something..

File: management-server/internal/config/types.go [hunk 5]
```diff
@@ -0,0 +1,28 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (VajiraPrabuddhaka) [inline management-server/internal/config/types.go:2]:
```suggestion
 *  Copyright (c) 2021, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: management-server/internal/config/parser.go (hunk 4 above)
Comment (VajiraPrabuddhaka) [inline management-server/internal/config/parser.go:40]:
```suggestion
	relativeConfigPath = "/conf/mgt_config.toml"
```

File: management-server/internal/config/parser.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (VajiraPrabuddhaka) [inline management-server/internal/config/parser.go:2]:
```suggestion
 *  Copyright (c) 2021, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: management-server/go.mod [hunk 7]
```diff
@@ -0,0 +1,24 @@
+module github.com/wso2/apk/APKManagementServer
```
Comment (VajiraPrabuddhaka) [inline management-server/go.mod:1]: shall we change the module name as `github.com/wso2/apk/management-server` like the same directory name.

File: management-server/internal/xds/server.go [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+	version := rand.Intn(maxRandomInt)
+	applications := apkmgt_application.Application{
+		Uuid: "apiUUID1",
+		Name: "name1",
+	}
+	newSnapshot, _ := wso2_cache.NewSnapshot(fmt.Sprint(version), map[wso2_resource.Type][]types.Resource{
+		wso2_resource.APKMgtApplicationType: {&applications},
+	})
+	apiCacheMutex.Lock()
+	apiCache.SetSnapshot(context.Background(), "mine", newSnapshot)
```
Comment (AmaliMatharaarachchi) [inline management-server/internal/xds/server.go:84]: read the node label from config

**Review by VajiraPrabuddhaka — APPROVED** [review]

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**